include/dxc/DXIL/DxilModule.h

```diff
@@ -404,6 +404,31 @@ class DxilModule {
                        std::unique_ptr<T> pRes);
   void LoadDxilSignature(const llvm::MDTuple *pSigTuple, DxilSignature &Sig,
                          bool bInput);
+
+public:
+  // Compute ShaderCompatInfo for all functions in module.
+  void ComputeShaderCompatInfo();
+
+  // Clear previous ShaderCompatInfo for recomputation.
+  void ClearShaderCompatInfo();
+
+  // ShaderCompatInfo tracks requirements per-function, subsequently merged into
+  // final entry function requirements.
+  struct ShaderCompatInfo {
+    unsigned minMajor = 6, minMinor = 0;
+    unsigned mask = ((unsigned)1 << (unsigned)DXIL::ShaderKind::Invalid) - 1;
+    ShaderFlags shaderFlags;
+    bool Merge(ShaderCompatInfo &other);
+  };
+
+  const ShaderCompatInfo *
+  GetCompatInfoForFunction(const llvm::Function *F) const;
+
+private:
+  typedef std::unordered_map<const llvm::Function *, ShaderCompatInfo>
```

Comment:
Consider using llvm::DenseMap instead of std::unordered_map?  ( I have been reading coding guidelines for the llvm project and it suggests that replacement for map usage.. I see other llvm:: objects are being used in this code, so I wanted to make this suggestion.

Reply:
I can do this, but there are several other pre-existing collections used in DxilModule for which the same argument stands.  This code uses a template already used elsewhere in DxilModule, and this was a move of existing code that used this map type, so perhaps a decision to change the collection types in DxilModule could be made separately?  What do you think?